web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php

```diff
@@ -155,6 +159,27 @@ public function proposalPageAttachments(&$page): void {
     }
   }
 
+  /**
+   * Implements hook_entity_presave().
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A proposal entity.
+   */
+  public function proposalEntityPresave(EntityInterface $entity): void {
+    if ('citizen_proposal' !== $entity->bundle()) {
+      return;
+    }
+    $proposalOriginal = $entity->original;
+    // Allow changing this value in settings.php $settings['sugggestion_period'] .
+    $periodLength = Settings::get('suggestion_period_length', self::SUGGESTION_PERIOD_LENGTH);
+    // If content is published.
+    if (1 === (int)$entity->status->value && 0 === (int)$proposalOriginal->status->value) {
```

Comment:
You could assert that `$entity` is a `NodeInterface` here and use `$entity->isPublished()`.

Reply:
Changed this to use isPublished method